### DISCUSSION TITLE
ESP32-C3 fails at IO wakeup  from light/deelpsleep.

### DISCUSSION THREAD

**Discussion by Kevinlee-512** [micropython/micropython#15744] [ESP32]
now, I have battery powered sensor project, everything including blue&wifi run well,  in order to get long battery life, need to put device
into deepsleep with timeout. However, immediate wakeup shall be expected by sensor trigger or button clicking.  Here is my key codes-  
...
key1=Pin(4,Pin.IN) 
key2=Pin(2,Pin.IN)
...
esp32.wake_on_ext1(pins = (key1,key2), level = esp32.WAKEUP_ANY_HIGH)  #initializing wake up
...
machine.deepsleep(hours)

I note that this codes work on ESP32.  Seems all the official ESP32-C3 firmware don't support,  I have searched for some similar question, still confused,  I wonder if any existing firmware to already resolve it ? or any idea to get working?  Highly appeciated at your help ...

**Comment by meirarmon** [comment]
the ESP32-C3 does not support EXT wakeup. I have an open PR (https://github.com/micropython/micropython/pull/17518) that adds `wake_on_gpio` for the C3, used like this (tested on my board):
```
import machine
import esp32
esp32.wake_on_gpio((machine.Pin(5),machine.Pin(2)), esp32.WAKEUP_ANY_HIGH)
machine.deepsleep()
```
